src/Couchbase.Lite.Shared/Query/QueryBase.cs

```diff
@@ -68,17 +65,48 @@ public Parameters Parameters
             set
             {
                 _queryParameters = value.Freeze();
-                Update();
+                SetParameters(_queryParameters.ToString());
             }
         }
 
+        internal string QueryExpression { get; set; }
+
+        [NotNull]
+        internal ThreadSafety ThreadSafety { get; set; } = new ThreadSafety();
+
+        internal SerialQueue DispatchQueue { get; } = new SerialQueue();
+
+        #endregion
+
+        #region Properties - Json Query Expression
+
+        protected bool Distinct { get; set; }
+
+        protected QueryDataSource FromImpl { get; set; }
```

Comment:
Why do all these need to be here?  These only apply to XQuery anyway...the point of having different classes becomes lost at this point.